plugins/spi/SPIOutput.cpp

```diff
@@ -486,6 +501,121 @@ uint8_t SPIOutput::P9813CreateFlag(uint8_t red, uint8_t green, uint8_t blue) {
   return ~flag;
 }
 
+
+void SPIOutput::IndividualAPA102Control(const DmxBuffer &buffer) {
+  // some detailed information on the protocol:
+  // https://cpldcpu.wordpress.com/2014/11/30/understanding-the-apa102-superled/
+  // Data-Struct
+  // StartFrame: 4 bytes = 32 bits zeros (APA102_START_FRAME_BYTES)
+  // LEDFrame: 1 byte FF ; 3 bytes color info (Blue, Green, Red)
+  // EndFrame: (n/2)bits; n = pixel_count
+
+  // calculate DMX-start-address
+  const unsigned int first_slot = m_start_address - 1;  // 0 offset
+
+  // only do something if at least 1 pixel can be updated..
+  if (buffer.Size() - first_slot < APA102_SLOTS_PER_PIXEL) {
+    OLA_INFO << "Insufficient DMX data, required " << APA102_SLOTS_PER_PIXEL
+             << ", got " << buffer.Size() - first_slot;
+    return;
+  }
+
+  // We always check out the entire string length, even if we only have data
+  // for part of it
+  const unsigned int output_length = APA102_START_FRAME_BYTES +
+                                (m_pixel_count * APA102_SPI_BYTES_PER_PIXEL);
+  uint8_t *output = m_backend->Checkout(m_output_number, output_length,
+                                      CalculateAPA102LatchBytes(m_pixel_count));
+  // only update SPI data if possible
+  if (!output) {
+    return;
+  }
+
```

Comment:
Please initialize the first 4 bytes here as well rather than relying on the defaults.